src/daft-core/src/array/ops/utf8.rs

```diff
@@ -725,6 +725,63 @@ impl Utf8Array {
         Ok(result)
     }
 
+    pub fn repeat<I>(&self, n: &DataArray<I>) -> DaftResult<Utf8Array>
+    where
+        I: DaftIntegerType,
+        <I as DaftNumericType>::Native: Ord,
+    {
+        let (is_full_null, expected_size) = parse_inputs(self, &[n])
+            .map_err(|e| DaftError::ValueError(format!("Error in repeat: {e}")))?;
+        if is_full_null {
+            return Ok(Utf8Array::full_null(
+                self.name(),
+                &DataType::Utf8,
+                expected_size,
+            ));
+        }
+        if expected_size == 0 {
+            return Ok(Utf8Array::empty(self.name(), &DataType::Utf8));
+        }
+
+        let self_iter = create_broadcasted_str_iter(self, expected_size);
+        let result = match n.len() {
+            1 => {
+                let n = n.get(0).unwrap();
+                let n: usize = NumCast::from(n).ok_or_else(|| {
+                    DaftError::ComputeError(format!(
+                        "Error in repeat: failed to cast rhs as usize {n}"
+                    ))
+                })?;
+                let n_iter = std::iter::repeat(n).take(expected_size);
+                let arrow_result = self_iter
+                    .zip(n_iter)
+                    .map(|(val, n)| Some(val?.repeat(n)))
+                    .collect::<arrow2::array::Utf8Array<i64>>();
```

Comment:
```suggestion
                let arrow_result = self_iter
                    .map(|val| Some(val?.repeat(n)))
                    .collect::<arrow2::array::Utf8Array<i64>>();
```

I don't think we need to create an iterator for `n`, and I think that I may have done this in a previous refactor for `str.left` and `str.right`. Can you check that this works and if so, help refactor for left and right as well?